internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -389,3 +395,18 @@ func expectedContainerEnv(containerSpec *egv1a1.KubernetesContainerSpec) []corev
 		return env
 	}
 }
+
+// caclulateMaxHeapSizeBytes calculates the maximum heap size in bytes as 80% of Envoy container memory limits.
+// In case no limits are defined '0' is returned, which means no heap size limit is set.
+func caclulateMaxHeapSizeBytes(envoyResourceRequirements *corev1.ResourceRequirements) uint64 {
+	if envoyResourceRequirements == nil || envoyResourceRequirements.Limits == nil {
```

Comment:
also lets make this method best effort
so it returns a *unit64, when we cannot compute the size lets return `nil`
and not add a max any `FixedHeapConfig` for that case

Reply:
looks like you already handle `0`, suggest changing the API signature to return a ptr instead so the caller can use `nil` as a way to make a decision

Reply:
`0` is an illegal value for `max_heap_size_bytes`, so it means it's `disabled`. It can be returned in 2 cases:
1. No memory limits are defined
2. Memory limits are `0`, which has the same affect as `1`

I don't see a value in changing the signature to ptr unless I missed something.

Reply:
cool if `0` is `disabled`, then sgtm, thats not clear from the API doc https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/resource_monitor/fixed_heap/v2alpha/fixed_heap.proto#config-resource-monitor-fixed-heap-v2alpha-fixedheapconfig

Reply:
Right, I verified it by defining 0 and got validation error from Envoy at runtime.